cfme/configure/settings.py

```diff
@@ -23,109 +25,173 @@ class TimeProfileAddForm(View):
     save_button = Button(VersionPick({Version.lowest(): 'Save', '5.9': 'Add'}))
     configuration = Dropdown('Configuration')
     table = Table("//div[@id='main_div']//table")
-    save_edit_button = Button('Save')
     cancel_button = Button('Cancel')
     help_block = Text("//span[contains(@class, 'help-block')]")
 
 
-class TimeprofileAddEntities(View):
+class TimeProfileAddEntities(View):
     breadcrumb = BreadCrumb()
     title = Text('//div[@id="main-content"]//h3')
 
 
 class TimeProfileAddFormView(BaseLoggedInPage):
-    timeprofile_form = View.nested(TimeProfileAddForm)
-    entities = View.nested(TimeprofileAddEntities)
+    time_profile_form = View.nested(TimeProfileAddForm)
+    entities = View.nested(TimeProfileAddEntities)
     mysetting = View.nested(MySettingsView)
 
 
-class Timeprofile(Updateable, Navigatable):
-    def __init__(self, description=None, scope=None, days=None, hours=None, timezone=None,
-                 appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        self.description = description
-        self.scope = scope
-        self.days = days
-        self.hours = hours
-        self.timezone = timezone
+class TimeProfileEditView(TimeProfileAddFormView):
+    save_edit_button = Button('Save')
 
-    def create(self, cancel=False):
-        view = navigate_to(self, 'All')
-        view.timeprofile_form.configuration.item_select('Add a new Time Profile')
-        view.timeprofile_form.fill({
-            'description': self.description,
-            'scope': self.scope,
-            'days': self.days,
-            'hours': self.hours,
-            'timezone': self.timezone,
-        })
-        if not cancel:
-            view.timeprofile_form.save_button.click()
-            view.flash.assert_message('Time Profile "{}" was saved'.format(self.description))
+
+@attr.s
+class TimeProfile(Updateable, BaseEntity):
+
+    description = attr.ib(default=None)
+    scope = attr.ib(default=None)
+    days = attr.ib(default=True)
+    hours = attr.ib(default=True)
+    timezone = attr.ib(default=None)
 
     def update(self, updates):
-        view = navigate_to(self, 'All')
-        rows = view.timeprofile_form.table
+        """
+        This method is used for updating the time_profile
+
+        Args:
+            updates: It the object of the time_profile that we need to update.
+        """
+
+        view = navigate_to(self, 'Edit')
+        rows = view.time_profile_form.table
         for row in rows:
             if row.description.text == self.description:
                 row[0].check()
-        view.timeprofile_form.configuration.item_select('Edit selected Time Profile')
-        changed = view.timeprofile_form.fill({
+        view.time_profile_form.configuration.item_select('Edit selected Time Profile')
```

Comment:
**Required**

It looks like this is repeating the action from the 'Edit' navigation destination's step function.  You've already navigated to the Edit page by selecting this configuration option, this should not be necessary here, or the loop to go through the rows checking